feature-libs/cart/base/components/clear-cart/clear-cart-dialog/clear-cart-dialog-component.service.ts

```diff
@@ -1,77 +1,69 @@
   ) {}
 
   clearActiveCart(): void {
-    this.isClearing$.next(true);
-    this.clearCart()
-      .pipe(take(1))
-      .subscribe(() => {
-        this.isClearing$.next(false);
-        this.onComplete();
+    this.activeCartFacade
+      .getEntries()
+      .pipe(
+        take(1),
+        tap((entries) => {
+          // Make copy and reverse entries[] to start at end of array
+          // since cart entries are shifted downwards with removeEntry()
+          entries
+            .slice()
+            .reverse()
+            .forEach((entry) => this.activeCartFacade.removeEntry(entry));
+        }),
+        switchMap(() => this.getClearingCartProgress()),
+        filter((inProgress) => !inProgress),
+        tap(() => this.closeDialog('Close dialog after cart cleared')),
+        switchMap(() =>
+          this.activeCartFacade
+            .getEntries()
+            .pipe(map((entries) => entries.length === 0))
+        ),
+        take(1)
+      )
+      .subscribe((status) => {
+        this.displayGlobalMessage(status);
       });
   }
 
-  clearCart(): Observable<boolean> {
-    return this.activeCartFacade.getEntries().pipe(
-      take(1),
-      tap((entries) => {
-        // Make copy and reverse entries[] to start at end of array
-        // since cart entries are shifted downwards with removeEntry()
-        entries
-          .slice()
-          .reverse()
-          .forEach((entry) => this.activeCartFacade.removeEntry(entry));
-      }),
-      switchMap(() => this.activeCartFacade.isStable()),
-      filter((data) => Boolean(data))
-    );
+  getClearingCartProgress(): Observable<boolean> {
+    return this.activeCartFacade.isStable().pipe(map((stable) => !stable));
   }
 
-  getClearingCartProgess(): Observable<boolean> {
-    return this.isClearing$.asObservable();
+  closeDialog(reason: string): void {
+    this.launchDialogService.closeDialog(reason);
   }
 
   /**
-   * Display succesful global message after clearing cart.
-   *
-   * @param success
+   * Display global message after clearing cart.
+   * By default, the message is displayed with the type `Success` only,
+   * a negative scenario has been handled in the occ layer already,
+   * but the status flag allows to recognize it and add a custom error message as well.
+   * @param status
    */
-  addSuccessGlobalMessage(): void {
-    this.globalMessageService.add(
-      { key: 'clearCart.cartClearedSuccessfully' },
-      GlobalMessageType.MSG_TYPE_CONFIRMATION
-    );
-  }
-
-  onComplete(): void {
-    this.closeDialog('Close dialog after cart cleared successfully');
-    this.activeCartFacade
-      .getEntries()
-      .pipe(take(1))
-      .subscribe((entries) => {
-        if (entries.length === 0) {
-          this.addSuccessGlobalMessage();
-        }
-      });
-  }
-
-  closeDialog(reason: string): void {
-    this.launchDialogService.closeDialog(reason);
+  protected displayGlobalMessage(status?: boolean): void {
+    if (status) {
```

Comment:
generally, this part of code seems an anti-pattern, but for this specified case I think it's possible to defend (possibility for customers to customize only notification area)